File: 748-largest-number-at-least-twice-of-others/largest-number-at-least-twice-of-others.java
class Solution {
    public int dominantIndex(int[] nums) {
        int max=nums[0];
        int r_i=0;
        for(int i=0;i<nums.length;i++){
            if(nums[i]>max){
                max=nums[i];
                r_i=i;
            }
        }
        for(int i=0;i<nums.length;i++){
            if(r_i!=i && max<2*nums[i]) 
            return -1;

        }
        return r_i;
    }
}

File: 748-largest-number-at-least-twice-of-others/largest-number-at-least-twice-of-others.cpp
class Solution {
public:
    int dominantIndex(vector<int>& nums) {
        int r_i=0;
        int l=nums[0];
        for(int i=0;i<nums.size();i++){
            if(nums[i]>l){
                r_i=i;
                l=nums[i];
            }
        }
        for(int i=0;i<nums.size();i++){
            if(i!=r_i && l<nums[i]*2){
                return -1;
            }
        }
        return r_i;
       
       
    }
};